util/src/network/connection.rs

```diff
@@ -205,7 +205,7 @@ impl Connection {
 	/// Update connection registration. Should be called at the end of the IO handler.
 	pub fn update_socket<Host: Handler>(&self, reg: Token, event_loop: &mut EventLoop<Host>) -> io::Result<()> {
 		trace!(target: "network", "connection reregister; token={:?}", reg);
-		event_loop.reregister( &self.socket, reg, self.interest, PollOpt::edge() | PollOpt::oneshot()).or_else(|e| {
+		event_loop.reregister( &self.socket, reg, self.interest, PollOpt::edge() /* | PollOpt::oneshot() */ ).or_else(|e| {  // TODO: oneshot is broken on windows
```

Comment:
No major issues, except some IO events are now generated earlier than required, and automatically. "oneshot" mode requires updating registration after each event handler and gives more precise control so that events don't queue up, but in our case it does not matter much since the handler is not a bottleneck.